// Generate with
// python -m grpc_tools.protoc -I. --python_out=. --grpc_python_out=. ./proto/songqueuer.proto

syntax = "proto3";

service SongQueueExchange {
    rpc AddSong(Song) returns (AddSongResponse) {}
    rpc DeleteSong(Song) returns (DeleteSongResponse) {}
    rpc PromoteSong(Song) returns (PromoteSongResponse) {}
}

message PromoteSongResponse {
    string response_text = 1;
}

message DeleteSongResponse {
    string response_text = 1;
}

message AddSongResponse {
    string response_text = 1;
}

message SongQueue {
    repeated Song song_queue = 1;
}

message Song {
    string id = 1;
    string title = 2;
    string file_name = 3;
    string requester_id = 4;
    string requester_name = 5;
}